### PR TITLE
ARGO-177 Ngi request support

### PR DESCRIPTION
**Pull request by makistsantekidis**

### CODE REVIEW

**Comment by makistsantekidis** [comment]
please review

@pkoro @kaggis @dpavlos

File: app/results/supergroup_test.go [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+	request, _ = http.NewRequest("GET", "/api/v2/results/Report_A/GROUP/GROUP_A?start_time=2015-06-20T12:00:00Z&end_time=2015-06-26T23:00:00Z", strings.NewReader(""))
+	request.Header.Set("x-api-key", suite.clientkey)
+	request.Header.Set("Accept", "application/json")
+
+	response = httptest.NewRecorder()
+
+	suite.router.ServeHTTP(response, request)
+
+	SuperGrouAvailabitiyJSON := `{
```
Comment (pkoro) [inline app/results/supergroup_test.go]: Typo

File: app/results/supergroup_test.go [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+// TestListAllSuperGroupAvailability test if daily results are returned correctly
+func (suite *SuperGroupAvailabilityTestSuite) TestListAllSuperGroupAvailability() {
+
+	request, _ := http.NewRequest("GET", "/api/v2/results/Report_A/GROUP?start_time=2015-06-20T12:00:00Z&end_time=2015-06-26T23:00:00Z&granularity=daily", strings.NewReader(""))
+	request.Header.Set("x-api-key", suite.clientkey)
+	response := httptest.NewRecorder()
+
+	suite.router.ServeHTTP(response, request)
+
+	SuperGroupAvailabitiyXML := ` <root>
```
Comment (pkoro) [inline app/results/supergroup_test.go]: typo

**Comment by pkoro** [comment]
Other than the couple of typos +1 from me. I think we should merge this one first and then deal with #109 as this one contains useful changes in the model.

File: app/results/Controller.go [hunk 3]
```diff
@@ -228,3 +334,133 @@ func MonthlyEndpointGroup(filter bson.M) []bson.M {
...
+			"reliability": bson.M{
+				"$divide": list{"$reliability", "$weights"}}},
+		},
+		{"$sort": bson.D{
+			{"report", 1},
+			{"supergroup", 1},
+			{"date", 1}},
+		}}
+
+	//query := []bson.M{{"$match": q}, {"$group": bson.M{"_id": bson.M{"dt": bson.D{{"$substr", list{"$dt", 0, 8}}}, "n": "$n", "ns": "$ns", "p": "$p"}, "a": bson.M{"$sum": bson.M{"$multiply": list{"$a", "$hs"}}}, 		"r": bson.M{"$sum": bson.M{"$multiply": list{"$r", "$hs"}}}, "hs": bson.M{"$sum": "$hs"}}}, {"$match": bson.M{"hs": bson.M{"$gt": 0}}}, {"$project": bson.M{"dt": "$_id.dt", "n": "$_id.n", "ns": "$_id.ns", "p": 		"$_id.p", "a": bson.M{"$divide": list{"$a", "$hs"}}, "r": bson.M{"$divide": list{"$r", "$hs"}}}}, {"$sort": bson.D{{"p", 1}, {"n", 1}, {"s", 1}, {"dt", 1}}}}
```
Comment (pkoro) [inline app/results/Controller.go]: remove comment here

File: app/results/Controller.go [hunk 4]
```diff
@@ -228,3 +334,133 @@ func MonthlyEndpointGroup(filter bson.M) []bson.M {
...
+	return query
+}
+
+// MonthlySuperGroup function to build the MongoDB aggregation query for monthly calculations
+func MonthlySuperGroup(filter bson.M) []bson.M {
+	//PROBABLY THIS LEADS TO THE SAME BUG WE RAN INTO WITH SITES. MUST BE INVESTIGATED!!!!!!!!!!!!
```
Comment (pkoro) [inline app/results/Controller.go]: add a TODO here

**Comment by pkoro** [comment]
@makistsantekidis note you are also missing documentation on this one. Other than that a monthly aggregation in the tests would also be nice.

File: app/results/supergroup_test.go [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+	response = httptest.NewRecorder()
+
+	suite.router.ServeHTTP(response, request)
+
+	SuperGrouAvailabilityJSON := `{
+   "root": [
+     {
+       "name": "GROUP_A",
+       "type": "GROUP",
+       "endpoints": [
```
Comment (pkoro) [inline app/results/supergroup_test.go]: Instead of `endpoints` better have `results` here.

**Comment by pkoro** [comment]
+1

**Pull request merged**